lib/tic_tac_toe.rb

```diff
@@ -0,0 +1,104 @@
+require_relative './player.rb'
+require_relative './board.rb'
+require_relative './custom_exception.rb'
+require_relative './validatable.rb'
+require 'colorize'
+class TicTacToe
+  attr_reader :total_match
+  def initialize
+    @total_match = 0
+    @players = []
+    @board = Board.new
+    @turn = 1
+    @used_symbol = {}
+    @draw = false
+  end
+
+  def play
+    @total_match += 1
+    @draw = false
+    @board.reset_board
+  end
+
+  def draw?
+    @draw
+  end
+
+  def display_board
+    @board.draw_board
+  end
+
+  def add_players(player_name, symbol)
+    raise CustomException, 'Can not add more that 2 players' if @players.size > 1
+    raise CustomException, 'Symbol already used try using another symbol' unless symbol_available?(symbol)
+
+    @players << Player.new(player_name, symbol)
+    @used_symbol[":#{symbol}"] = true
+  end
+
+  def change_turn
+    @turn = @turn == 2 ? 1 : 2
+    get_player(@turn).name + '\'s Turn'
+  end
+
+  def ended?()
+    @draw = @board.total_moves > 8
+  end
+
+  def make_move(tile)
+    raise CustomException, 'Invalid Input, Valid Move should be digit between 1-9' unless validate_move(tile)
+
+    tile = tile.to_i
+    @board.mark_tile(tile, get_player(@turn).symbol)
+  end
+
+  def player_won?
+    if @board.won?(get_player(@turn).symbol)
+      get_player(@turn).increase_win
+      return true
+    end
+    false
+  end
+
+  def display_winner_message
+    puts <<-HEARDOC
+
+    ************************************************************
+    *                    Congratulations                        *
+                      #{get_player(@turn).name}  has won
+
+    HEARDOC
+    display_statstics
+  end
```

Comment:
* [ ] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- As it was suggested by @majovanilla you can return a String variable instead and print it in your main.rb file

Reply:
**Fixed**, it must have slipped :smiley: